File: Element/sum_equal_zero.cxx
// Subarray which sum equal zero ~ O(N²) time complexity

#include <bits/stdc++.h>
using namespace std;


class Solution{
	public:
	static bool sum_zero(vector<int> &arr){
		for (auto i = 0; i < arr.size(); i++){
			int sum = 0;
			for(auto j = i+1; j < arr.size(); j++){
				sum += arr[j];
				if(sum == 0)
					return true;
			}
		}
		return false;
	}
};
int main(){
	ios_base::sync_with_stdio(0);
	vector<int> test = {4, 2, -3, 1, 6}; // True
	cout << Solution::sum_zero(test);
}

File: Element/sum_equal_zero.py
class Solution(object):
    def __init__(self,arr):
        self.arr = arr
    def sum_zero(self,arr)->bool:
        self.n = len(self.arr)
        for i in range(self.n):
            self.sum = 0
            for j in range(i+1,self.n):
                self.sum += self.arr[j]
                if(self.sum == 0):
                    return True
        return False
    

if __name__=='__main__':
    test = [4,2,-3,1,6]
    s = Solution(test)
    print(s.sum_zero(test)) # True (2,-3,1)
